src/BoundsInference.cpp

```diff
@@ -159,18 +158,17 @@ class BoundsInference : public IRMutator {
                 // figure out what those dimensions are, and just have all
                 // stages but the last use the bounds for the last stage.
                 vector<bool> always_pure_dims(func.args().size(), true);
-                const std::vector<UpdateDefinition> &updates = func.updates();
-                for (size_t i = 0; i < updates.size(); i++) {
+                for (UpdateDefinition i : func.updates()) {
                     for (size_t j = 0; j < always_pure_dims.size(); j++) {
-                        const Variable *v = updates[i].args[j].as<Variable>();
+                        const Variable *v = i.args[j].as<Variable>();
                         if (!v || v->name != func.args()[j]) {
                             always_pure_dims[j] = false;
                         }
                     }
                 }
 
-                if (stage < (int)func.updates().size()) {
-                    size_t stages = func.updates().size();
+                if (stage < func.num_update_definitions()) {
```

Comment:
I added these to mirror the existence of these functions (e.g. num_update_definitions) in the Func class, but I'm fine with going either way.